src/Raven.Client/Documents/Changes/DatabaseChanges.cs

```diff
@@ -220,6 +229,61 @@ public IChangesObservable<DocumentChange> ForDocumentsInCollection<TEntity>()
             return ForDocumentsInCollection(collectionName);
         }
 
+        public IChangesObservable<CounterChange> ForAllCounters()
+        {
+            var counter = GetOrAddConnectionState("all-counters", "watch-counters", "unwatch-counters", null);
+
+            var taskedObservable = new ChangesObservable<CounterChange, DatabaseConnectionState>(
+                counter,
+                notification => true);
+
+            return taskedObservable;
+        }
+
+        public IChangesObservable<CounterChange> ForCounter(string counterName)
+        {
+            if (string.IsNullOrWhiteSpace(counterName))
+                throw new ArgumentException("Value cannot be null or whitespace.", nameof(counterName));
+
+            var counter = GetOrAddConnectionState($"counter/{counterName}", "watch-counter", "unwatch-counter", counterName);
+
+            var taskedObservable = new ChangesObservable<CounterChange, DatabaseConnectionState>(
+                counter,
+                notification => string.Equals(counterName, notification.Name, StringComparison.OrdinalIgnoreCase));
+
+            return taskedObservable;
+        }
+
+        public IChangesObservable<CounterChange> ForCounterOfDocument(string documentId, string counterName)
+        {
+            if (string.IsNullOrWhiteSpace(documentId))
+                throw new ArgumentException("Value cannot be null or whitespace.", nameof(documentId));
+            if (string.IsNullOrWhiteSpace(counterName))
+                throw new ArgumentException("Value cannot be null or whitespace.", nameof(counterName));
+
+            var counter = GetOrAddConnectionState($"counter/{counterName}/document/{documentId}", "watch-document-counter", "unwatch-document-counter", string.Join(Separator, documentId, counterName));
```

Comment:
I would actually expect it to be document, then counter, since counter is inside doc.

Reply:
Not sure what I follow?

Reply:
`counter/{counterName}/document/{documentId}` vs: 
`document/{documentId}/counter/{counterName}`

Reply:
I wanted to be consistent with `ForCounterOfDocument` -> `counter/of/document`. This is used only internally in ConcurrentDict.

Reply:
Okay, this is an in memory only value, anyway.

Reply:
fixed